composer/callbacks/image_monitor.py

```diff
@@ -0,0 +1,194 @@
+        logger.data_batch({key: table})
+
+    def before_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'input' and state.timestamp.get(self.interval.unit).value % self.interval.value == 0:
+            self._log_inputs(state, logger, 'Images/Train')
+
+    def eval_before_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'input' and state.eval_timestamp.get(TimeUnit.BATCH).value == 0:
+            self._log_inputs(state, logger, 'Images/Eval')
+
+    def before_loss(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'segmentation' and state.timestamp.get(
+                self.interval.unit).value % self.interval.value == 0:
+            self._log_segmented_inputs(state, logger, 'Images/Train')
+
+    def eval_after_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'segmentation' and state.eval_timestamp.get(TimeUnit.BATCH).value == 0:
+            self._log_segmented_inputs(state, logger, 'Images/Eval')
+
+
+def _make_input_images(inputs: torch.Tensor, num_images: int):
+    import wandb
+    if inputs.shape[0] < num_images:
+        num_images = inputs.shape[0]
+    images = inputs[0:num_images].data.cpu().permute(0, 2, 3, 1).numpy()
+    table = wandb.Table(columns=['Image'])
+    for image in images:
+        img = wandb.Image(image)
+        table.add_data(img)
+    return table
+
+
+def _make_segmentation_images(inputs: torch.Tensor,
+                              targets: torch.Tensor,
+                              outputs: torch.Tensor,
+                              num_images: int,
+                              shift_targets: bool = False):
+    import wandb
+    if min([inputs.shape[0], targets.shape[0], outputs.shape[0]]) < num_images:
+        num_images = min([inputs.shape[0], targets.shape[0], outputs.shape[0]])
+    images = inputs[0:num_images].data.cpu().permute(0, 2, 3, 1).numpy()
+    targets = targets[0:num_images].data.cpu().numpy()
+    outputs = outputs[0:num_images]
+    # Convert outputs to segmentation masks. Assume channels are first dim
+    outputs = outputs[0:num_images]
+    outputs = outputs.argmax(dim=1).cpu().numpy()
+    # Shift targets such that negative values are mapped to 0
+    if shift_targets:
+        targets += 1
+        targets[targets < 0] = 0
+        outputs += 1
+    table = wandb.Table(columns=['Image'])
+    for image, target, prediction in zip(images, targets, outputs):
+        mask = {'ground truth': {'mask_data': target}, 'prediction': {'mask_data': prediction}}
+        img_mask_pair = wandb.Image(image, masks=mask)
+        table.add_data(img_mask_pair)
+    return table
+
+
+def _check_for_image_format(data: torch.Tensor) -> bool:
```

Comment:
Maybe here you could add the check for numpy array and PIL Image. Also the checking for multiple tensors can be checked here?

Reply:
I added the check, but decided to add it earlier in the code.